pyro/nn/auto_reg_nn.py

```diff
@@ -104,39 +102,66 @@ class AutoRegressiveNN(nn.Module):
     """
     An implementation of a MADE-like auto-regressive neural network.
 
-    Reference:
-    MADE: Masked Autoencoder for Distribution Estimation [arXiv:1502.03509]
-    Mathieu Germain, Karol Gregor, Iain Murray, Hugo Larochelle
+    Example usage:
+    >>> x = torch.randn(100, 10)
+    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1])
+    >>> p = arn(x)  # 1 parameters of size (100, 10)
+    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1, 1])
+    >>> m, s = arn(x) # 2 parameters of size (100, 10)
+    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1, 5, 3])
+    >>> a, b, c = arn(x) # 3 parameters of sizes, (100, 1, 10), (100, 5, 10), (100, 3, 10)
```

Comment:
one might expect sizes (100, 10) (100, 50) (100, 30) ?

Reply:
why this choice?

Reply:
I thought that you would typically reshape it like that anyway before you use it in another class. Also it avoids any confusion about whether the ordering is by (param_dim, input_dim) or (input_dim, param_dim). What do you think?

Reply:
seems reasonable